app/scripts/lib/multichain-api/wallet-requestPermissions.js

```diff
@@ -65,119 +68,125 @@ async function requestPermissionsImplementation(
   const [requestedPermissions] = params;
   delete requestedPermissions[Caip25EndowmentPermissionName];
```

Comment:
🤔 why is this necessary? We shouldn't really expect this... oh is this preventing the approval/confirmation from being confused if this is somehow on the request in addition to a legacy typed request?

Reply:
I'm unsure if there is an issue caused by passing the CAIP-25 permission into the approval controller. My goal was to prevent it from being passed into the permissions controller since we are handling the approval and grant manually in this handler